spec.bs

```diff
@@ -2859,6 +2864,175 @@ This specification defines two [=policy-controlled features=] identified by the
+:: An [=origin=]. Matches the origin that served the captured [:Ad-Auction-Signals:] header.
+: <dfn>ad slot</dfn>
+:: A [=string=]. Matches the `adSlot` key of the JSON dictionaries in the top level array of the
+  [:Ad-Auction-Signals:] value.
+
+</dl>
+
+<h3 id=direct-from-sellers-signals-struct>Direct from seller signals</h3>
+A <dfn>direct from seller signals</dfn> is a [=struct=] with the following [=struct/items=]:
+
+<dl dfn-for="direct from seller signals">
+: <dfn>seller signals</dfn>
+:: Null or a [=string=].
+  Opaque JSON data passed to both buyers' and the seller's [=script runners=].
+: <dfn>auction signals</dfn>
+:: Null or a [=string=].
+  Opaque JSON data passed to the seller's [=script runner=].
+: <dfn>per buyer signals</dfn>
+:: A [=map=] whose [=map/keys=] are [=origins=] and whose [=map/values=] are [=strings=].
+  [=map/Keys=] are buyers and must be valid HTTPS origins. [=map/Values=] are opaque JSON data
+  passed to corresponding buyer's [=script runner=].
+
+</dl>
+
+Each [=traversable navigable=] has a <dfn>captured ad auction headers</dfn>, which is a [=map=]
+whose [=map/keys=] are [=direct from seller signals keys=] and whose [=map/values=] are [=direct from seller signals=].
+</div>
+
+<div algorithm="fetch auction headers patch">
+The following will be added to [[Fetch#http-extensions]]:
+
+<h3 id=sec-ad-auction-fetch-header>`Sec-Ad-Auction-Fetch` header</h3>
+
+The <dfn http-header><code>Sec-Ad-Auction-Fetch</code></dfn> request header is an optional
+[=structured header=] with of type [=structured header/boolean=]. If equal to `?1`, the user agent will remove any
+[:Ad-Auction-Signals:] from the response returned to {{WindowOrWorkerGlobalScope/fetch()}} -- the
+[:Ad-Auction-Signals:] value will instead only be used in Protected Audiences auctions. [:Sec-Ad-Auction-Fetch:]
+will only be set if the {{WindowOrWorkerGlobalScope/fetch()}} call was made with the `{adAuctionHeaders: true}`
+option.
+
+<h3 id=ad-auction-signals-header>`Ad-Auction-Signals` header</h3>
+
+The <dfn http-header><code>Ad-Auction-Signals</code></dfn> response header provides value
+of a JSON array of dictionaries, each with an `adSlot` key. Protected Audience non-component, component, and
+top-level auctions may specify which signals to load by the `adSlot` key.
+</div>
+
+<div algorithm="fetch auction signals response patch">
+The following step will be added to the [=http-network-or-cache fetch=] algorithm, before step "Return |response|.":
+
+1. If |response| is not null, then run [=update captured headers=] with [=this=]'s
+  [=relevant global object=]'s [=associated Document's=] [=node navigable's=]
+  [=traversable navigable's=] [=captured ad auction headers=], |response|'s
+  [=response/header list=], and <var ignore>request</var>'s [=request/URL=]'s [=url/origin=].
+
+</div>
+
+<div algorithm="fetch update cached headers patch">
+The following algorithm will be added to the [=http-network-or-cache fetch=] algorithm:

```

Comment:
How is this whole algorithm going to be added to another already-defined algorithm?

Reply:
Fixed